src/Symfony/Bundle/FrameworkBundle/Resources/public/csrf_double_submit_controller.js

```diff
@@ -0,0 +1,34 @@
+document.addEventListener('submit', function (event) {
+    const csrfField = event.target.querySelector('input[data-controller="symfony--framework-bundle--csrf-double-submit"]');
+
+    if (!csrfField) {
+        return;
+    }
+
+    let csrfCookie = csrfField.getAttribute('data-symfony--framework-bundle--csrf-double-submit-cookie');
+
+    if (!csrfCookie) {
+        csrfField.setAttribute('data-symfony--framework-bundle--csrf-double-submit-cookie', csrfCookie = csrfField.value);
+    }
+
+
+    if (!/^[-a-zA-Z0-9_]+$/.test(csrfCookie)) {
+        return
+    }
+
+    let csrfToken = btoa(String.fromCharCode.apply(null, (window.crypto || window.msCrypto).getRandomValues(new Uint8Array(18))));
+    csrfField.value = csrfToken;
+    let cookie = csrfCookie + '=' + csrfToken + '; path=/; samesite=strict';
+
+    document.cookie = window.location.protocol === 'https:' ? '__Host-' + cookie + '; secure' : cookie;
```

Comment:
As it's a new feature, should we really offer a security feature in non-secure context ? :|

Reply:
Yes, why not. We're not on a mission to force the internet to upgrade the internet. We're on a mission to help build something useful on it.